File: app/src/main/java/org/jzs/mybaseapp/common/widget/MyDialog.java
package org.jzs.mybaseapp.common.widget;

import android.app.Dialog;
import android.content.Context;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.TextView;

import org.jzs.zhihudaily.R;


/**
 * 通用dialog
 */
public class MyDialog implements OnClickListener {

    private Dialog dialog;

    private Context context;

    private DialogCallBack mCallBack;

    public Dialog showDialog(Context context, String content) {
        this.context = context;
        dialog = new Dialog(context, R.style.custom_dialog);
        dialog.setContentView(R.layout.dialog_my);
        dialog.findViewById(R.id.btn_other_cancel).setOnClickListener(this);
        dialog.findViewById(R.id.btn_other_ok).setOnClickListener(this);
        ((TextView) dialog.findViewById(R.id.tv_other_text)).setText(content);
        dialog.show();
        return dialog;
    }

    public void setTitle(String str) {
        ((TextView) dialog.findViewById(R.id.tv_title)).setText(str);
    }

    public void setCancleGone() {
        dialog.findViewById(R.id.btn_other_cancel).setVisibility(View.GONE);
    }

    public void setSureGone() {
        dialog.findViewById(R.id.btn_other_ok).setVisibility(View.GONE);
    }

    public void setSureText(String str) {
        ((Button) dialog.findViewById(R.id.btn_other_ok)).setText(str);
    }

    public void setCancleText(String str) {
        ((Button) dialog.findViewById(R.id.btn_other_cancel)).setText(str);
    }

    public void setCancleBackground(int res) {
        ((Button) dialog.findViewById(R.id.btn_other_cancel)).setBackground(ContextCompat.getDrawable(context, res));
    }

    public void setCallBack(MyDialog.DialogCallBack dialogCallBack) {

        this.mCallBack = dialogCallBack;
    }

    @Override
    public void onClick(View v) {

        // TODO Auto-generated method stub
        switch (v.getId()) {
            case R.id.btn_other_cancel:
                if (mCallBack != null) {
                    mCallBack.cancelHandle();
                }
                dialog.dismiss();
                break;
            case R.id.btn_other_ok:

                if (mCallBack != null) {
                    mCallBack.handle();
                }
                dialog.dismiss();
                break;
        }
    }

    public interface DialogCallBack {

        void handle();

        void cancelHandle();
    }
}

File: SkUtils/src/main/java/org/jzs/skutils/dialog/MyDialog.kt
package org.jzs.skutils.dialog

import android.app.Dialog
import android.content.Context
import android.support.v4.content.ContextCompat
import android.view.View
import android.widget.Button
import android.widget.TextView
import org.jzs.skutils.R


/**
 * 通用dialog
 */
class MyDialog : View.OnClickListener {

    private var dialog: Dialog? = null

    private var context: Context? = null

    private var mCallBack: DialogCallBack? = null

    fun showDialog(context: Context): Dialog {
        this.context = context
        dialog = Dialog(context, R.style.custom_dialog)
        dialog!!.setContentView(R.layout.dialog_my)
        dialog!!.findViewById<View>(R.id.btn_cancel).setOnClickListener(this)
        dialog!!.findViewById<View>(R.id.btn_ok).setOnClickListener(this)
        dialog!!.show()
        return dialog as Dialog
    }

    fun setContent(str: String) {
        (dialog!!.findViewById<View>(R.id.tv_content) as TextView).text = str
    }

    fun setTitle(str: String) {
        (dialog!!.findViewById<View>(R.id.tv_title) as TextView).text = str
    }

    fun setCancleGone() {
        dialog!!.findViewById<View>(R.id.btn_cancel).visibility = View.GONE
    }

    fun setSureGone() {
        dialog!!.findViewById<View>(R.id.btn_ok).visibility = View.GONE
    }

    fun setSureText(str: String) {
        (dialog!!.findViewById<View>(R.id.btn_ok) as Button).text = str
    }

    fun setCancleText(str: String) {
        (dialog!!.findViewById<Button>(R.id.btn_cancel))!!.text = str
    }

    fun setCancleBackground(res: Int) {
        dialog!!.findViewById<Button>(R.id.btn_cancel).setBackgroundColor(ContextCompat.getColor(context!!, res))
    }

    fun setCallBack(dialogCallBack: DialogCallBack) {
        this.mCallBack = dialogCallBack
    }

    override fun onClick(v: View) {
        val i = v.id
        if (i == R.id.btn_cancel) {
            if (mCallBack != null) {
                mCallBack!!.cancel()
            }
            dialog!!.dismiss()

        } else if (i == R.id.btn_ok) {
            if (mCallBack != null) {
                mCallBack!!.sure()
            }
            dialog!!.dismiss()

        }
    }

    interface DialogCallBack {
        fun sure()

        fun cancel()
    }
}
